version: '3.8'

services:
  user-service:
    build:
      context: ./user-service
      dockerfile: Dockerfile
    ports:
      - "8080:8080"
    networks:
      - eda-network
    depends_on:
      rabbitmq:
        condition: service_healthy
      db:
        condition: service_started

  email-service:
    build:
      context: ./email-service
      dockerfile: Dockerfile
    ports:
      - "8081:8081"
    networks:
      - eda-network
    depends_on:
      rabbitmq:
        condition: service_healthy
      user-service:
        condition: service_started

  db:
    image: postgres:16
    ports:
      - "5432:5432"
    environment:
      - POSTGRES_DB=DB
      - POSTGRES_USER=USER
      - POSTGRES_PASSWORD=PASSWORD
    volumes:
      - ./eda-data:/var/lib/postgresql/data
    networks:
      - eda-network

  rabbitmq:
    image: rabbitmq:3-management
    ports:
      - "5672:5672"
      - "15672:15672"
    environment:
      RABBITMQ_DEFAULT_USER: RABBITMQ_USERNAME
      RABBITMQ_DEFAULT_PASS: RABBITMQ_PASSWORD
    networks:
      - eda-network
    healthcheck:
      test: ["CMD", "rabbitmq-diagnostics", "ping"]
      interval: 5s
      timeout: 5s
      retries: 5

networks:
  eda-network:
    driver: bridge

volumes:
  eda-data:
